app/uploaders/validate_pdf.rb

```diff
@@ -11,10 +13,9 @@ def validate_pdf(file)
     temp_file = file.tempfile
     return unless temp_file.readpartial(4) == '%PDF'
 
-    PDF::Reader.new(temp_file).info
-  rescue PDF::Reader::MalformedPDFError
-    raise CarrierWave::UploadError, 'PDF is missing an end of file marker'
-  rescue PDF::Reader::EncryptedPDFError
-    raise CarrierWave::UploadError, 'PDF is encrypted'
+    pdf = Origami::PDF.read temp_file, decrypt: false
+    raise CarrierWave::UploadError, 'PDF is encrypted' if pdf.encrypted?
+  rescue Origami::InvalidPDFError
+    raise CarrierWave::UploadError, 'PDF is invalid'
```

Comment:
Can we get more detailed on the responses here? More specific error handling could result in easier debugging in the future.

Reply:
I've updated these to be a bit more descriptive.